spec/javascripts/unit/core/channels/presence_channel_spec.js

```diff
@@ -214,10 +235,25 @@ describe("PresenceChannel", function() {
           var callback = jasmine.createSpy("callback");
           channel.bind("something", callback);
 
-          channel.handleEvent("something", 9);
+          channel.handleMessage({
+            event: "something",
+            data: 9
+          });
 
           expect(callback).toHaveBeenCalledWith(9);
         });
+        it("should include user_id in metadata", function() {
+          var callback = jasmine.createSpy("callback");
+          channel.bind("client-something", callback);
+
+          channel.handleMessage({
+            event: "client-something",
+            data: 9,
+            user_id: "abc-def",
+          });
+
+          expect(callback).toHaveBeenCalledWith(9, {user_id: "abc-def"});
```

Comment:
Doesn't the callback know about the event name?

Reply:
Nope. The callback has been executed by that event.